concent_api/core/validation.py

```diff
@@ -334,3 +335,25 @@ def validate_compute_task_def(compute_task_def: message.tasks.ComputeTaskDef) ->
             )
 
     validate_scene_file(extra_data['scene_file'])
+
+
+def validate_if_given_object_is_expected_enum_instance(
+    custom_object: enum.Enum,
+    base_enum: type,
+    allow_none_as_instance: bool = False,
+) -> None:
+
+    if allow_none_as_instance is True and custom_object is None:
+        return
+
+    if allow_none_as_instance is False and custom_object is None:
+        raise GolemMessageValidationError(
+            f'{custom_object} is None, it should be {base_enum} instance',
+            error_code=ErrorCode.MESSAGE_WRONG_FIELDS
```

Comment:
Maybe, but just maybe, this would be worth a different error code ? Because the problem in this case is rather with value, not field. We have code `MESSAGE_VALUE_NOT_ALLOWED`. But please wait for @cameel opinion on this. Choosing proper error code is tricky.

Reply:
I think this is not the right place to choose an error code. This function does not know in which context it's being called. How does it even know that the enum comes from inside a golem message?

This the same problem as in #394. The function should just say if it's correct or not and the caller should be responsible for raising the error.

Reply:
Good point, then it has to be different exception because `GolemMessageValidationError` requires error code.

Reply:
1. I think reusing built-in errors like `TypeError` is not a good idea because almost always they're caused by bugs. We want to be able to discern them from other errors that may indicate failures. For example if you make a mistake in your validation and Python raises `TypeError`. your function will interpret it as a failed validation, which is not true (it's a bug in our code, not a failed validation caused by invalid input).
2. I think that it's not necessary to discern between `None` and other types. They both indicate something that not's a value from the enum.